cmd/state-tray/main.go

```diff
@@ -0,0 +1,73 @@
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/ActiveState/cli/cmd/state-tray/internal/open"
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/getlantern/systray"
+	"github.com/gobuffalo/packr"
+)
+
+func main() {
+	systray.Run(onReady, onExit)
+}
+
+func onReady() {
+	box := packr.NewBox("assets")
+	systray.SetIcon(box.Bytes("icon.ico"))
+	systray.SetTooltip("ActiveState State Tool")
+
+	mAbout := systray.AddMenuItem("About State Tool", "Information about the State Tool")
+
+	systray.AddSeparator()
+
+	mDoc := systray.AddMenuItem("Documentation", "Open State Tool Docs")
+
+	mPlatform := systray.AddMenuItem("ActiveState Platform", "")
+	mLearn := mPlatform.AddSubMenuItem("Learn", "Open ActiveState blog")
+	mSupport := mPlatform.AddSubMenuItem("Support", "Open support page")
+	mAccount := mPlatform.AddSubMenuItem("Account", "Open your account page")
+
+	systray.AddSeparator()
+
+	mQuit := systray.AddMenuItem("Exit", "")
+
+	for {
+		select {
+		case <-mAbout.ClickedCh:
+			logging.Debug("About event")
+			err := open.Prompt("state --version")
+			if err != nil {
+				handleError(errs.Wrap(err, "Could not open command prompt"))
```

Comment:
This is awkward. This function should just be returning errors. Then the calling function can have one single call to `handleError`, or just handle the error itself. The point is to unify the error handling, not to send it into a black hole.

Reply:
I'm not convinced that logging and error and printing it to `stderr` is a black hole. However, I see what you mean about unifying the error handling. I've wrapped this function and handled the errors in one place.

Reply:
It's a black hole from the perspective of the function. I'd rather be handling errors "the go way" and then have one piece of code handle the reporting. By just throwing errors into this function at any point that you would otherwise returning an error it become a bit of a "black hole".